server/src/storage/mod.rs

```diff
@@ -78,8 +80,9 @@ pub struct StorageEngine {
 }
 
 impl StorageEngine {
-    pub fn new() -> Self {
+    pub fn new(storage_folder_path: String) -> Self {
```

Comment:
Since the [refactor-storage-engine](https://github.com/ModelarData/ModelarDB-RS/tree/dev/refactor-storage-engine) branch is built on top of this branch and changes a significant amount of the code i have made this [change](https://github.com/ModelarData/ModelarDB-RS/commit/0ea31a4c378c394f6b8a83588b338d40536e168d) in that branch instead. I would prefer to avoid continuing work on code that is not merged yet but this was not possible in this case.